apps/trading/pages/index.page.tsx

```diff
@@ -32,6 +32,14 @@ export function Index() {
   // The default market selected in the platform behind the overlay
   // should be the oldest market that is currently trading in continuous mode(i.e. not in auction).
   const { data, error, loading } = useQuery<MarketsLanding>(MARKETS_QUERY);
+  if (data && !error && !loading) {
+    const marketId = marketList(data)[0]?.id;
+    window.history.replaceState(
```

Comment:
the purpose was to make the link sharable and consistent with the oldest traded market found. 

after lots of things that I've tried, this is the most efficient way of maintaining index functionality, but shallow route without reloading, and without messing up with the windows history - so that going back won't redirect to '/'.

Reply:
question (blocking): Did you look at using the useRouter component from next? See here: https://nextjs.org/docs/api-reference/next/router.

Altering the window history directly in the render function is probably a bad idea. I would suggest putting this logic in a useEffect hook and using the replace function from useRouter.

```
const { replace } = useRouter()

useEffect(() => {
  if (marketId) {
    replace(`/markets/${marketId}`)
  }
}, [marketId])
```

Reply:
With the above, I think you'll probably want to render the <LandingDialog /> on the market page. 

We end up with no index page at all and we just use it to redirect where we need to go, for example:
```
function Index() {
 const { data, error, loading} = useQuery(...)

  useEffect(() => {
    ... calculate desired market and redirect
  }, [data])
  
  return null
}
```

Reply:
so that was the first approach I've tried.
this requires to render the modal on the market page only if it's the first launch of the app (not the first render of the market page). 

 and for this I needed to maintain some global state to show the select market modal only on landing. 
 
 I've also tried using hooks with localstorage, but that does not give me the desired behaviour.

Reply:
I tried initially to use only `next/router` methods such as push and replace:
```
      replace(
        `....`,
        `/markets/${marketId}?portfolio=orders&trade=orderbook&chart=candles`,
        { shallow: true }
      );
```
And even with shallow set to true it would still do a proper redirect to the market page, there is a small glitch, and the modal closes as it does not know it is the first pop up

Reply:
also - what I have here is subject to change - this is because in my next ticket (started to work on it already) I will launch another modal to select the markets (and it will open the grid that we have on the `/markets` page currently).  

https://github.com/vegaprotocol/frontend-monorepo/issues/456

Reply:
> question (blocking): Did you look at using the useRouter hook from next? See here: https://nextjs.org/docs/api-reference/next/router.
> 
> Altering the window history directly in the render function is probably a bad idea. I would suggest putting this logic in a useEffect hook and using the replace function from useRouter.
> 
> ```
> const { replace } = useRouter()
> 
> useEffect(() => {
>   if (marketId) {
>     replace(`/markets/${marketId}`)
>   }
> }, [marketId])
> ```
I tried initially to use only next/router methods such as push and replace:

      replace(
        `....`,
        `/markets/${marketId}?portfolio=orders&trade=orderbook&chart=candles`,
        { shallow: true }
      );
And even with shallow set to true it would still do a proper redirect to the market page, there is a small glitch, and the modal closes as it does not know it is the first pop up

Reply:
> so that was the first approach I've tried. this requires to render the modal on the market page only if it's the first launch of the app (not the first render of the market page).
> 
> and for this I needed to maintain some global state to show the select market modal only on landing (I don't think keeping global state is necessary right now, but I might find out that it is while working on the next ticket #456).
> 
> I've also tried using hooks with localstorage, but that does not give me the desired behaviour.

so that was the first approach I've tried.
this requires to render the modal on the market page only if it's the first launch of the app (not the first render of the market page).

and for this I needed to maintain some global state to show the select market modal only on landing (I don't think keeping global state is necessary right now, but I might find out that it is while working on the next ticket https://github.com/vegaprotocol/frontend-monorepo/issues/456).

I've also tried using hooks with localstorage, but that does not give me the desired behaviour.

Reply:
also - what I have here is subject to change - this is because in my next ticket (started to work on it already) I will launch another modal to select the markets (and it will open the grid that we have on the /markets page currently).

https://github.com/vegaprotocol/frontend-monorepo/issues/456

Reply:
> With the above, I think you'll probably want to render the on the market page.
> 
> We end up with no index page at all and we just use it to redirect where we need to go, for example:
> 
> ```
> function Index() {
>  const { data, error, loading} = useQuery(...)
> 
>   useEffect(() => {
>     ... calculate desired market and redirect
>   }, [data])
>   
>   return null
> }
> ```

so that was the first approach I've tried.
this requires to render the modal on the market page only if it's the first launch of the app (not the first render of the market page).

and for this I needed to maintain some global state to show the select market modal only on landing (I don't think keeping global state is necessary right now, but I might find out that it is while working on the next ticket https://github.com/vegaprotocol/frontend-monorepo/issues/456).

I've also tried using hooks with localstorage, but that does not give me the desired behaviour.

Reply:
also - what I have here is subject to change - this is because in my next ticket (started to work on it already) I will launch another modal to select the markets (and it will open the grid that we have on the /markets page currently).

https://github.com/vegaprotocol/frontend-monorepo/issues/456